crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -1649,11 +1653,23 @@ struct InboundGroupSessionIndexedDbObject {
     /// "refer to the `needs_backup` property". See:
     /// https://github.com/element-hq/element-web/issues/26892#issuecomment-1906336076
     backed_up_to: i32,
+
+    /// The curve key of the device that send us this room key, base64-encoded.
+    curve_key: Option<String>,
+
+    /// The type of the [`SenderData`] within this session, converted to a u8
+    /// from [`SenderDataType`].
+    sender_data_type: Option<u8>,
```

Comment:
answer: serde gives us a string (`"UnknownDevice"`, `"DeviceInfo"`); allowing ints is essentially what https://github.com/serde-rs/serde/pull/2525 is about.

It might be worth stopping to consider which is actually better here. A number is smaller (though experience suggests that they take up 8 bytes anyway), and presumably quicker to index. A string is maybe "safer" in the face of changes, though since we're assigning the enum discriminants explicitly, I can't really imagine there being any confusion.

I think this is fine, though what I'll do is change `InboundGroupSessionIndexedDbObject::new` to take a `SenderDataType` rather than a `u8`, to limit the number of places we are dealing with raw numbers.